Modules/HMPID/src/HmpidTaskMatches.cxx

```diff
@@ -0,0 +1,185 @@
+// // granted to it by virtue of its status as an Intergovernmental Organization
+// // or submit itself to any jurisdiction.
+//
+
+///
+/// \file   HmpidTaskMatches.cxx
+/// \author Nicola Nicassio, Giacomo Volpe
+///
+
+#include <TCanvas.h>
+#include <TH1.h>
+#include <TH2.h>
+
+#include "QualityControl/QcInfoLogger.h"
+#include "HMPID/HmpidTaskMatches.h"
+#include <Framework/InputRecord.h>
+#include <Framework/InputRecordWalker.h>
+#include <Framework/InputSpec.h>
+#include "GlobalTrackingWorkflowHelpers/InputHelper.h"
+#include "ReconstructionDataFormats/TrackTPCITS.h"
+#include "DataFormatsTPC/TrackTPC.h"
+#include "DataFormatsGlobalTracking/RecoContainerCreateTracksVariadic.h"
+#include "ReconstructionDataFormats/TrackParametrization.h"
+#include "DetectorsBase/Propagator.h"
+
+#include "DataFormatsHMP/Cluster.h"
+#include "ReconstructionDataFormats/MatchInfoHMP.h"
+
+#ifdef WITH_OPENMP
+#include <omp.h>
+#endif
+
+namespace o2::quality_control_modules::hmpid
+{
+
+HmpidTaskMatches::HmpidTaskMatches() : TaskInterface() {}
+
+HmpidTaskMatches::~HmpidTaskMatches()
+{
+  ILOG(Info, Support) << "destructor called" << ENDM;
+  for (int iCh = 0; iCh < 7; iCh++) {
+    delete mMatchInfoResidualsXTrackMIP[iCh];
+    delete mMatchInfoResidualsYTrackMIP[iCh];
+    delete mMatchInfoChargeClusterMIP[iCh];
+    delete mMatchInfoChargeClusterPhotons[iCh];
+    delete mMatchInfoClusterMIPMap[iCh];
+    delete mMatchInfoThetaCherenkovVsMom[iCh];
+  }
+}
+
+void HmpidTaskMatches::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Debug, Devel) << "initialize TaskMatches" << ENDM;
+
+  mDataRequest = std::make_shared<o2::globaltracking::DataRequest>();
+  mDataRequest->requestTracks(mSrc, false);
+
+  // getJsonParameters();
+
+  BookHistograms();
+
+  // publish histograms (Q: why publishing in initalize?)
+  for (int iCh = 0; iCh < 7; iCh++) {
+
+    getObjectsManager()->startPublishing(mMatchInfoResidualsXTrackMIP[iCh]);
+    getObjectsManager()->startPublishing(mMatchInfoResidualsYTrackMIP[iCh]);
+    getObjectsManager()->startPublishing(mMatchInfoChargeClusterMIP[iCh]);
+    getObjectsManager()->startPublishing(mMatchInfoChargeClusterPhotons[iCh]);
+    getObjectsManager()->startPublishing(mMatchInfoClusterMIPMap[iCh]);
+    getObjectsManager()->setDefaultDrawOptions(mMatchInfoClusterMIPMap[iCh], "colz");
+    getObjectsManager()->setDisplayHint(mMatchInfoClusterMIPMap[iCh], "colz");
+    getObjectsManager()->startPublishing(mMatchInfoThetaCherenkovVsMom[iCh]);
+  }
+
+  ILOG(Info, Support) << "START DOING QC HMPID Matches" << ENDM;
+
+  // here do the QC
+}
+
+void HmpidTaskMatches::startOfActivity(const Activity& /*activity*/)
+{
+  ILOG(Debug, Devel) << "startOfActivity" << ENDM;
+  reset();
+}
+
+void HmpidTaskMatches::startOfCycle()
+{
+  ILOG(Debug, Devel) << "startOfCycle" << ENDM;
+}
+
+void HmpidTaskMatches::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  ILOG(Info, Support) << "monitorData" << ENDM;
```

Comment:
Never log from `monitorData` : it is called many times a second and floods the logs.

Reply:
Removed!